static/src/javascripts/projects/common/modules/check-dispatcher.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+import config from 'lib/config';
+import { allEmailCanRun, listCanRun } from 'common/modules/email/run-checks';
+import emailArticle from 'common/modules/email/email-article';
+import clash from 'common/modules/experiments/ab-test-clash';
+import trackAdRender from 'commercial/modules/dfp/track-ad-render';
+import commercialFeatures from 'commercial/modules/commercial-features';
+import { checks } from './check-mediator-checks';
+import { resolveCheck, waitForCheck } from './check-mediator';
+
+const someCheckPassed = results => results.includes(true);
```

Comment:
As we discussed IRL, I don't think there's any distinction in type annotation requirements between public and private functions. It's a case of balancing time you want to spend versus the [piece of mind that Flow gives us](https://github.com/guardian/frontend/wiki/So-you-want-to-ES6%3F#specific-way) 😅